src/dshould/basic.d

```diff
@@ -0,0 +1,328 @@
+            }
+            else
+            {
+                check(
+                    got !is expected,
+                    format(": expected %s%s", refInfo, expected.quote),
+                    format(", but got %s", got.quote),
+                    file, line
+                );
+            }
+        }
+        else
+        {
+            const refInfo = isReferenceType ? "same reference as " : "";
+
+            static if (is(T == typeof(null)))
+            {
+                check(got is null, ": expected null", format(", but got %s", got.quote), file, line);
+            }
+            else
+            {
+
+                check(
+                    got is expected,
+                    format(": expected %s%s", refInfo, expected.quote),
+                    format(", but got %s", got.quote),
+                    file, line
+                );
+            }
+        }
+    }
+}
+
+auto equal(Should)(Should should)
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.allowOnlyWordsBefore!(["not", "be", "greater", "smaller"], "equal");
+
+    return should.addWord!"equal";
+}
+
+void equal(Should, T)(Should should, T expected, string file = __FILE__, size_t line = __LINE__)
+if (isInstanceOf!(ShouldType, Should) && !should.hasWord!"approximately")
+{
+    should.equal.numericCheck(expected, file, line);
+}
+
+auto greater(Should)(Should should)
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.allowOnlyWordsBefore!(["not", "be", "equal", "smaller"], "greater");
+    should.requireWord!("be", "greater");
+
+    return should.addWord!"greater";
+}
+
+void greater(Should, T)(Should should, T expected, string file = __FILE__, size_t line = __LINE__)
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.greater.numericCheck(expected, file, line);
+}
+
+auto smaller(Should)(Should should)
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.allowOnlyWordsBefore!(["not", "be", "equal", "greater"], "smaller");
+    should.requireWord!("be", "smaller");
+
+    return should.addWord!"smaller";
+}
+
+void smaller(Should, T)(Should should, T expected, string file = __FILE__, size_t line = __LINE__)
+if (isInstanceOf!(ShouldType, Should))
+{
+    should.smaller.numericCheck(expected, file, line);
+}
+
+void numericCheck(Should, T)(Should should, T expected, string file, size_t line)
+if (isInstanceOf!(ShouldType, Should))
+{
+    import std.format : format;
+
+    with (should)
+    {
+        enum notPart = hasWord!"not" ? "!" : "";
+        enum equalPart = hasWord!"equal" ? "==" : "";
+        enum equalPartShort = hasWord!"equal" ? "=" : "";
+        enum smallerPart = hasWord!"smaller" ? "<" : "";
+        enum greaterPart = hasWord!"greater" ? ">" : "";
+
+        auto got = should.got();
+
+        enum isFloating = __traits(isFloating, typeof(got));
+
+        static if (isFloating)
+        {
+            enum combinedWithoutEqual = notPart ~ smallerPart ~ greaterPart;
```

Comment:
there is no (longer a) comparison `!<` or `!>`

Reply:
when did that happen :o